temboardagent/notification.py

```diff
@@ -1,64 +1,116 @@
         if type(username) == bytes:
             username = username.decode('utf-8')
         if type(message) == bytes:
             message = message.decode('utf-8')
+
+        self.time = now()
+        self.date = date
         self.username = username
         self.message = message
 
 
 class NotificationMgmt():
 
+    @classmethod
+    def bootstrap(self, config):
+        db_path = os.path.join(config.temboard.home, 'core.db')
+        with sqlite3.connect(db_path) as conn:
+            c = conn.cursor()
+            c.execute(
+                dedent("""
+                    CREATE TABLE IF NOT EXISTS action_logs (
+                        time REAL,
+                        date TEXT,
+                        username TEXT,
+                        message TEXT
+                    )
+                """)
+            )
+            c.execute(
+                dedent("""
+                    CREATE INDEX IF NOT EXISTS idx_action_logs_time
+                    ON action_logs(time)
+                """)
+            )
+
     @classmethod
     def push(self, config, notification):
         try:
-            # Notifications are stored in a "sliding" queue.
-            q = Queue(file_path='%s/notifications.q' % (
-                                config.temboard['home']),
-                      max_size=10 * 1024 * 1024,  # 10MB
-                      overflow_mode='slide')
-
-            # Push the notification in the queue.
-            q.push(Message(content=json.dumps({
-                'date': notification.date,
-                'username': notification.username,
-                'message': notification.message})))
-        except (Exception) as e:
-            raise NotificationError('Can not push new notification: %s' %
-                                    e.message)
+
+            db_path = os.path.join(config.temboard.home, 'core.db')
+            with sqlite3.connect(db_path) as conn:
+                c = conn.cursor()
+                c.execute(
+                    "INSERT INTO action_logs VALUES (?, ?, ?, ?)",
+                    (notification.time, notification.date,
+                     notification.username, notification.message)
+                )
+                # Purge action_logs, we want to keep only the last 100 messages
+                c.execute(
+                    dedent("""
+                        DELETE FROM action_logs
+                        WHERE time NOT IN (
+                            SELECT time
+                            FROM action_logs
+                            ORDER BY time DESC
+                            LIMIT 100
+                        )
+                    """)
+                )
+
+        except sqlite3.Error as e:
+            logger.exception(str(e))
+            raise NotificationError('Can not push new notification')
 
     @classmethod
     def get_last_n(self, config, n):
+
+        limit = " LIMIT %s" % str(n) if int(n) > -1 else ""
```

Comment:
In `"%s" % str(n)`, the `str()` is superfluous because `%s` will perform the string conversion. (nit)